### ISSUE TITLE
Cannot Force Quit / Remove from Memory Large Built JS File

### ISSUE THREAD

**Issue by joevbruno** [Microsoft/vscode#25452]
- VSCode Version: Version 1.11.2 (1.11.2)
- OS Version: MacOS Sierra 10.12.4 

I had a large build JS file open (from Webpack bundle) in VSCode with unsaved changes. I deleted the file from the command line as part of a build-script. Thus, the file no longer existed on the file system. Thus, when attempting to save and close all tabs, the editor prompted me to save this file; however, it then hung, and I was forced to force-quit VSCode. Now, every time I open VSCode, it reopens the file and immediately hangs. I've tried leaving for hours, force quitting, deleting vscode and reinstall (soft - I don't want to blow away all my extensions if possible), restarting, cleaning app memory and caches, etc. The file always re-opens whenever I open the folder where my project is. I tried renaming the folder and now the file opens in a separate window at least so I can resume working, but the file is still always around as an unsaved file (and again, its deleted from the file system OS). How can I delete it from VSCode's memory.

Steps to Reproduce:

1. Open a large Webpack Built JS file. 
2. Delete it from the command line outside the editor.
3. Attempt to close the editor - prompt for unsaved changes save
4. Editor hangs
5. Force quit
6. Stuck.

**Comment by alexandrudima** [comment]
@joevbruno This sounds like hot exit kicking in, coupled with a file that triggers some sort of bug.

Is it possible for you to share the file with us so we can try to investigate why VS Code would hang when opening it?

`$HOME/Library/Application Support/Code/User/Backups` would contain the hot exit files under OSX I believe. 

You could rename that folder to Backups2 and perhaps share the relevant file in there with us, we should not hang when opening a file, even if it is large. I'm suspecting there could be some issue in the JS grammar we are shipping with that is triggered by a code pattern in that specific file and that might cause the freeze. 

Alternatively, you could try installing an extension that offers a different JS grammar, such as "JavaScript Atom Grammar" and see if opening the file still hangs.

**Comment by bpasero** [comment]
@joevbruno what are you picking in the dialog when you want to close the editor of the large file?

**Comment by joevbruno** [comment]
@bpasero @alexandrudima I found the file in `Application Support/Code/Backups`. It's 4.2MB in size. Honestly, I don't feel super comfortable sharing the file because it was a server-side dev webpack build (isomorphic app) that is un-minified and contains secret keys for our apis and third-party services.  I know that is not a huge help. If it was personal project it would be different. 

@bpasero the problem is I cannot hit any button. Doesn't matter what I choose, nothing works. I've hit every button, but I can only force-quit from OSX. 

Interestingly enough I have "JavaScript Atom Grammar" installed. Enabling or disabling it seems to have no effect. 

I deleted the file from backups and restart. The file no longer opens when I upon up my project, so thank you for that!
